File: woodroid/src/main/java/me/gilo/woodroid/Builder.java
package me.gilo.woodroid;

import me.gilo.woodroid.data.ApiVersion;

public class Builder {
    private String siteUrl;
    private ApiVersion apiVerion;
    private String consumerKey;
    private String consumerSecret;


    public Builder() {
    }

    public Builder setSiteUrl(String siteUrl) {
        this.siteUrl = siteUrl;
        return this;
    }

    public Builder setApiVersion(ApiVersion apiVerion) {
        this.apiVerion = apiVerion;
        return this;
    }

    public Builder setConsumerKey(String consumerKey) {
        this.consumerKey = consumerKey;
        return this;
    }

    public Builder setConsumerSecret(String consumerSecret) {
        this.consumerSecret = consumerSecret;
        return this;
    }


    public Woocommerce build() {
        return new Woocommerce(siteUrl, apiVerion, consumerKey, consumerSecret);
    }
}

File: woodroid/src/main/java/me/gilo/woodroid/Builder.kt
package me.gilo.woodroid

import me.gilo.woodroid.data.ApiVersion


class Builder {
    private lateinit var siteUrl: String
    private lateinit var apiVerion: ApiVersion
    private lateinit var consumerKey: String
    private lateinit var consumerSecret: String

    fun setSiteUrl(siteUrl: String): Builder {
        this.siteUrl = siteUrl
        return this
    }

    fun setApiVersion(apiVerion: ApiVersion): Builder {
        this.apiVerion = apiVerion
        return this
    }

    fun setConsumerKey(consumerKey: String): Builder {
        this.consumerKey = consumerKey
        return this
    }

    fun setConsumerSecret(consumerSecret: String): Builder {
        this.consumerSecret = consumerSecret
        return this
    }


    fun build(): Woocommerce {
        return Woocommerce(siteUrl, apiVerion, consumerKey, consumerSecret)
    }
}